microsoft/testsuites/network/sriov.py

```diff
@@ -310,9 +328,263 @@ def sriov_add_max_nics_validation(
     def sriov_provision_with_max_nics_validation(
         self, environment: Environment
     ) -> None:
-        self._sriov_basic_test(environment)
+        vm_nics = self._initialize_nic_info(environment)
+        self._sriov_basic_test(environment, vm_nics)
+
+    @TestCaseMetadata(
+        description="""
+        This case verify VM works well during remove and load sriov modules.
+
+        Steps,
+        1. Provision VM with max network interfaces with enabling accelerated network.
+        2. Do the basic sriov testing.
+        3. Remove sriov module, check network traffic through synthetic nic.
+        4. Load sriov module, check network traffic through VF.
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            min_count=2,
+            min_nic_count=8,
+            network_interface=schema.NetworkInterfaceOptionSettings(
+                data_path=schema.NetworkDataPath.Sriov,
+            ),
+        ),
+    )
+    def sriov_reload_modules_validation(self, environment: Environment) -> None:
+        vm_nics = self._initialize_nic_info(environment)
+        self._sriov_basic_test(environment, vm_nics)
+        module_in_used = ""
+        for node in environment.nodes.list():
+            module_in_used = self._remove_module(node)
+        self._sriov_vf_connection_test(environment, vm_nics, remove_module=True)
+        for node in environment.nodes.list():
+            self._load_module(node, module_in_used)
+        vm_nics = self._initialize_nic_info(environment)
+        self._sriov_vf_connection_test(environment, vm_nics)
+
+    @TestCaseMetadata(
+        description="""
+        This case verify below two kernel patches.
+        1. hv_netvsc: Sync offloading features to VF NIC
+           https://github.com/torvalds/linux/commit/68622d071e555e1528f3e7807f30f73311c1acae#diff-007213ba7199932efdb096be47d209a2f83e4d425c486b3adaba861d0a0c80c5 # noqa: E501
+        2. hv_netvsc: Allow scatter-gather feature to be tunable
+           https://github.com/torvalds/linux/commit/b441f79532ec13dc82d05c55badc4da1f62a6141#diff-007213ba7199932efdb096be47d209a2f83e4d425c486b3adaba861d0a0c80c5 # noqa: E501
+
+        Steps,
+        1. Change scatter-gather feature on synthetic nic,
+         verify the the feature status sync to the VF dynamically.
+        2. Disable and enable sriov,
+         check the scatter-gather feature status keep consistent in VF.
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            min_count=2,
+            network_interface=schema.NetworkInterfaceOptionSettings(
+                nic_count=3,
```

Comment:
What's the reason it needs exact 3 nics? How about limit the minimum size only?

Reply:
there are two cases in lisav2, one is to use 1 nic, the other is to use 3 nics, I think use one case with 3 nics is enough?

Reply:
How about change it to min 3 and max 6? Then add a comments about why the number is chose.

Reply:
actually >=1 is okay.

Reply:
Is it changed already?

Reply:
not yet, I didn't figure out which value should I specify